src/main/java/bwapi/BWClient.java

```diff
@@ -7,58 +7,112 @@
  */
 public class BWClient {
     private final BWEventListener eventListener;
-    private final boolean debugConnection;
-    private EventHandler handler;
+    private BotWrapper botWrapper;
+    private Client client;
+    private PerformanceMetrics performanceMetrics;
 
     public BWClient(final BWEventListener eventListener) {
-        this(eventListener, false);
-    }
-
-    /**
-     * @param debugConnection set to `true` for more explicit error messages (might spam the terminal).
-     *                        `false` by default
-     */
-    public BWClient(final BWEventListener eventListener, final boolean debugConnection) {
         Objects.requireNonNull(eventListener);
-        this.debugConnection = debugConnection;
         this.eventListener = eventListener;
     }
 
     /**
      * Get the {@link Game} instance of the currently running game.
+     * When running in asynchronous mode, this is the game from the bot's perspective, eg. potentially a previous frame.
      */
     public Game getGame() {
-        return handler == null ? null : handler.getGame();
+        return botWrapper == null ? null : botWrapper.getGame();
+    }
+
+    /**
+     * @return JBWAPI performance metrics.
+     */
+    public PerformanceMetrics getPerformanceMetrics() {
+        return performanceMetrics;
+    }
+
+    /**
+     * @return The number of frames between the one exposed to the bot and the most recent received by JBWAPI.
+     * This tracks the size of the frame buffer except when the game is paused (which results in multiple frames arriving with the same count).
+     */
+    public int framesBehind() {
+        return botWrapper == null ? 0 : Math.max(0, client.clientData().gameData().getFrameCount() - getGame().getFrameCount());
+    }
+
+    /**
+     * For internal test use.
+     */
+    Client getClient() {
+        return client;
     }
 
+    /**
+     * Start the game with default settings.
+     */
     public void startGame() {
-        startGame(false);
+        BWClientConfiguration configuration = new BWClientConfiguration();
+        startGame(configuration);
     }
 
     /**
      * Start the game.
      *
      * @param autoContinue automatically continue playing the next game(s). false by default
      */
+    @Deprecated
```

Comment:
This overload is replaced by one which takes a BWClientConfiguration to indicate thhe autoContinue setting.